fix(migrations): don't copy unmigrated declarations into imports array

crisbeto opened the pull request:
Currently the migration is set up to assume that any elements that exist in a `declarations` array will be converted to standalone and copied into the `imports` array, however that might be incorrect for some special cases like the root component.

These changes rework the declaration merging logic so that they take all the declarations being migrated into account.

devversion commented on packages/core/schematics/ng-generate/standalone-migration/to-standalone.ts:
```diff
@@ -131,94 +132,116 @@ function getComponentImportExpressions(
+      }
+    } else {
+      // Otherwise create a spread that will be copied into the `imports`.
+      declarationsToCopy.push(ts.factory.createSpreadElement(declarationsProp.initializer));
```
So for spread cases we may still move stuff incorrectly. I guess that's coming up rarely like usual, but wanted to make sure it's a deliberate decision.

devversion reviewed (approved):
lgtm

crisbeto committed:
fix(migrations): don't copy unmigrated declarations into imports array

Currently the migration is set up to assume that any elements that exist in a `declarations` array will be converted to standalone and copied into the `imports` array, however that might be incorrect for some special cases like the root component.

These changes rework the declaration merging logic so that they take all the declarations being migrated into account.

dylhunn commented:
This PR was merged into the repository by commit 83895578488bd35c7e47609f092907eb0f53f435.
